topology/probes/k8s.go

```diff
@@ -0,0 +1,236 @@
+/*
+ * Copyright 2016 IBM Corp.
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ *
+ */
+
+package probes
+
+import (
+	"github.com/skydive-project/skydive/logging"
+	"github.com/skydive-project/skydive/topology/graph"
+	"github.com/skydive-project/skydive/config"
+	"github.com/FlorianOtel/client-go/kubernetes"
+	"github.com/FlorianOtel/client-go/pkg/api/v1"
+	"github.com/FlorianOtel/client-go/tools/clientcmd"
+	"github.com/FlorianOtel/client-go/pkg/types"
+	"strings"
+)
+
+type K8SMessage struct {
+	msgType string
+	obj     interface{}
+}
+
+type K8SProbe struct {
+	Graph			*graph.Graph
+	nodeUpdaterChan 	chan graph.Identifier
+	messagesUpdaterChan 	chan K8SMessage
+	uid 			[]types.UID
+	clientset		*kubernetes.Clientset
+}
+
+// sync between pods (k8s.k8sPods) as reported by k8s and skydive netns IDs (k8s.NetnsIDs)
+func (k8s *K8SProbe) isK8S(nodeName string) bool {
+	for _,uid := range k8s.uid{
+		// match name with nodeName
+		logging.GetLogger().Debugf("Compare nodeName %s with uid %s \n", nodeName, string(uid))
+		if strings.Contains(nodeName, string(uid)) {
+			// update k8sPods
+			logging.GetLogger().Debugf("Found match! nodeName %s with uid %s \n", nodeName, string(uid))
+			return true
+		}
+	}
+	logging.GetLogger().Debugf("No match is found for nodeName %s \n", nodeName)
+	return false
+}
+
+// gets pods from k8s
+func (k8s *K8SProbe) Getk8sPods() {
+
+	pods, err := k8s.clientset.Core().Pods("").List(v1.ListOptions{})
+	if err != nil {
+		panic(err.Error())
```

Comment:
panic here is not a good idea, better to log error adding a recover mechanism otherwise if there is a network issue Skydive will crash